price_security/models/account_payment_term.py

```diff
@@ -2,10 +2,17 @@
 # For copyright and license notices, see __manifest__.py file in module root
 # directory
 ##############################################################################
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class AccountPaymentTerm(models.Model):
     _inherit = "account.payment.term"
 
     sequence = fields.Integer()
+
+    @api.model
+    def create(self, vals):
+        if "sequence" in vals:
```

Comment:
No seria un not in? 
así estamos pisando solo si la defini, podria quedar en False.
En general odoo define solo un default y despues agrega un al modelo un order  tipo 
`_order = "sequence DESC, id ASC" `
Y porque no con un compute, no se llamaba correctamente? no ahorariamos el post_init_hook